posts/2024-05-06-jakarta-nosql-in-action-meet-jnopo-game.adoc

```diff
@@ -0,0 +1,828 @@
+
+* The game identification;
+* A tied attribute, to know if the game has ended as a tied one;
+* The player A and its movement;
+* The player B and its movement;
+* The winner player and its movement, if the game has ended as a non-tied one.
+* The loser player and its movement, if the game has ended as a non-tied one.
+
+Next step: let`s create the `GameMatchCapturer` bean that will be responsible to capture the emitted `GameOver` state and then store the game match results into the database:
+
+[source, java]
+----
+package org.jakartaee.sample.model;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.event.Observes;
+import jakarta.inject.Inject;
+import org.jakartaee.sample.game.GameOver;
+
+@ApplicationScoped
+public class GameMatchCapturer {
+
+    public void captureAndPersist(@Observes GameOver gameOver){
+        // put the persistence logic here...
+    }
+
+}
+
+----
+
+At this point, let's instantiate a new `GameMatch` entity instance, populate it from `GameOver` data and then store it into the database by using the `Playoffs` component.
+
+The required `Playoff` component, which is a repository implementation offered by Jakarta Data specification implementation, Eclipse JNoSQL, will be injected by CDI using the `@Inject` and `@Database(DOCUMENT)` annotations:
+
+[source, java]
+----
+package org.jakartaee.sample.model;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.event.Observes;
+import jakarta.inject.Inject;
+import org.eclipse.jnosql.mapping.Database;
+import org.eclipse.jnosql.mapping.DatabaseType;
+import org.jakartaee.sample.game.GameOver;
+
+@ApplicationScoped
+public class GameMatchCapturer {
+
+    @Inject
+    @Database(DatabaseType.DOCUMENT)
+    Playoffs playoffs;
+
+    public void captureAndPersist(@Observes GameOver gameOver){
+
+            var gameMatch  = new GameMatch(
+                    gameOver.gameId(),
+                    PlayerInfo.of(gameOver.playerAInfo()),
+                    PlayerInfo.of(gameOver.playerBInfo()),
+                    gameOver.winnerInfo().map(PlayerInfo::of).orElse(PlayerInfo.NOBODY),
+                    gameOver.winnerInfo().map(PlayerInfo::of).orElse(PlayerInfo.NOBODY),
+                    gameOver.isTied()
+            );
+
+            playoffs.add(gameMatch);
+    }
+
+}
+----
+
+Now, let's get started the  https://openliberty.io/[Open Liberty] runtime to figure out if the game match results are being stored on the MongoDB:
+
+* On Linux/Mac machines, perform the following command:
++
+[source, bash]
+----
+./mvnw liberty:dev
+----
+
+* On Windows machines, perform the following command:
++
+[source, powershell]
+----
+mvnw.cmd liberty:dev
+----
+
```

Comment:
Does this need to be started up again if we did not shut it down before? We're asking them to run in dev mode, so any changes should have already been picked up.